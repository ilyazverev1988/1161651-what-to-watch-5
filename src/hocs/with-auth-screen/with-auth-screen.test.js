import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withActiveItem from "../with-active-item/with-active-item";


const noop = () => {};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveItem(MockComponent);

describe(`Should MoviePage render correctly`, () => {
  it(`With error Autorization`, () => {
    const tree = renderer.create(
      <MockComponentWrapped
        onSubmit={noop}
        errorAuthorization={`We canâ€™t recognize this email and password combination. Please try again.`}
      >
        <React.Fragment/>
      </MockComponentWrapped>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Without error Autorization`, () => {
    const tree = renderer.create(
      <MockComponentWrapped
        onSubmit={noop}
        errorAuthorization={``}
      >
        <React.Fragment/>
      </MockComponentWrapped>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
