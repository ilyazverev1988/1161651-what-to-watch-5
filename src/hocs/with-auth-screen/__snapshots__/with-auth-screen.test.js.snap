// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MoviePage render correctly With error Autorization 1`] = `<div />`;

exports[`Should MoviePage render correctly Without error Autorization 1`] = `<div />`;
